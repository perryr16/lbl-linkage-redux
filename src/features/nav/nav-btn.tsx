import React from 'react';
import {BrowserRouter, Link, Switch, Route} from 'react-router-dom'


interface Props {
   pageName: string,
   path: string,
   nextComponent: any,
}

export const NavBtn: React.FC<Props> = (props) => {
   const {pageName, path, nextComponent} = props
   return (
      <BrowserRouter>
      <div>
         <Link data-testid='a-tag' to={path}>{pageName}</Link>
      </div>
      <Switch>
         <Route path={path}>
            {nextComponent}
         </Route>
      </Switch>
      </BrowserRouter>
   );
}