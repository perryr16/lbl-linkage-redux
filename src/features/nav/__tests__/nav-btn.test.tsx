import React from 'react';
import {render, cleanup} from '@testing-library/react';
import '@testing-library/jest-dom';
import {NavBtn} from '../nav-btn';


afterEach(cleanup);

it('renders', () => {
  const {asFragment} = render(
    <NavBtn
      pageName='test name'
      path='/'
      nextComponent='next component'
    />
  )
  expect(asFragment()).toMatchSnapshot();
});

it('has passes props', () => {
  const {getByTestId, getByText} = render(
    <NavBtn
      pageName='test name'
      path='/'
      nextComponent='next component'
    />
  )
  expect(getByTestId("a-tag")).toHaveTextContent('test name')
})