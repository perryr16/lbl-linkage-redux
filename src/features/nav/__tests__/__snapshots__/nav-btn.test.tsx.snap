// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders 1`] = `
<DocumentFragment>
  <div>
    <a
      data-testid="a-tag"
      href="/"
    >
      test name
    </a>
  </div>
  next component
</DocumentFragment>
`;
